aten/src/ATen/native/vulkan/api/Allocator.h

```diff
@@ -2,11 +2,19 @@
 
 #include <ATen/native/vulkan/api/Common.h>
 
+#ifdef DEBUG
+  #define VMA_DEBUG_LOG(format, ...)  \
+    do {                              \
+      printf(format, ##__VA_ARGS__);  \
+      printf("\n");                   \
+    } while(false)
```

Comment:
Is it some common pattern to do this in the do-while(false) cycle?
In which cases will it run more than once?

Reply:
Yes, it is a common pattern.  It is done so as to allow macros to be used safely in all place including places where only a single statement is allowed.  This wiki has a good explanation (https://wiki.sei.cmu.edu/confluence/display/c/PRE10-C.+Wrap+multistatement+macros+in+a+do-while+loop) and contains other implementations that might come to mind but are flawed.  By the way, the rest of the advice in there looks solid as well.

Reply:
Forgot to mention that it does not run more than once.  Execution runs through the body and terminates after the first run since the condition is false.